--- a/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -493,8 +493,7 @@
 }

 function shouldDisplayAutoChannel(hwtype, configChannel, runtimeChannel) {
-	return configChannel == 'auto' || runtimeChannel == 'auto' ||
-		((isQcaWifiHwtype(hwtype) || hwtype == 'mt_dbdc') && isUnsetChannelValue(configChannel));
+    return true;
 }

 function isPlausibleTxPowerValue(txpower, hwtype) {
@@ -1880,7 +1879,7 @@
 		const ch = +channel;
 		const channelMeta = this.getChannelMeta(band, channel);
 		const bandName = String(band || '');
-		const restrictWide = (band == '5g' && !(ch > 0 && ch <= 100));
+		const restrictWide = false;

 		const filtered = [];

