--- a/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -493,8 +493,7 @@
 }

 function shouldDisplayAutoChannel(hwtype, configChannel, runtimeChannel) {
-	return configChannel == 'auto' || runtimeChannel == 'auto' ||
-		((isQcaWifiHwtype(hwtype) || hwtype == 'mt_dbdc') && isUnsetChannelValue(configChannel));
+    return true;
 }

 function isPlausibleTxPowerValue(txpower, hwtype) {
@@ -1880,7 +1879,7 @@
 		const ch = +channel;
 		const channelMeta = this.getChannelMeta(band, channel);
 		const bandName = String(band || '');
-		const restrictWide = (band == '5g' && !(ch > 0 && ch <= 100));
+		const restrictWide = false;

 		const filtered = [];

===================================================================
--- a/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
@@ -277,10 +277,6 @@
 		spi-tx-bus-width = <4>;
 		spi-rx-bus-width = <4>;

-		mediatek,nmbm;
-		mediatek,bmt-max-ratio = <1>;
-		mediatek,bmt-max-reserved-blocks = <64>;
-
 		partitions: partitions {
 			compatible = "fixed-partitions";
 			#address-cells = <1>;
@@ -343,6 +339,13 @@
 			partition@600000 {
 				label = "ubi";
 				reg = <0x0600000 0x7000000>;
+				compatible = "linux,ubi";
+
+				volumes {
+					ubi_rootdisk: ubi-volume-fit {
+						volname = "fit";
+					};
+				};
 			};

 			partition@7600000 {
===================================================================
--- a/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
@@ -22,6 +22,9 @@
 		CI_KERNPART="linux"
 		nand_do_upgrade "$1"
 		;;
+	xiaomi,mi-router-ax3000t)
+	    fit_do_upgrade "$1"
+        ;;
 	bananapi,bpi-r3|\
 	bananapi,bpi-r4|\
 	bananapi,bpi-r4-2g5|\
===================================================================
--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -1012,12 +1012,16 @@
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
   PAGESIZE := 2048
+  DEVICE_PACKAGES := kmod-mt7915e kmod-mt7981-firmware mt7981-wo-firmware
   KERNEL_IN_UBI := 1
-  DEVICE_PACKAGES := kmod-mt7981-firmware mt7981-wo-firmware
-  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  UBOOTENV_IN_UBI := 1
+  IMAGES := sysupgrade.itb
+  KERNEL := kernel-bin | gzip
+  IMAGE/sysupgrade.itb := append-kernel | \
+	fit gzip $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb external-static-with-rootfs | append-metadata
 endef
 TARGET_DEVICES += xiaomi_mi-router-ax3000t

 define Device/xiaomi_mi-router-wr30u
   DEVICE_VENDOR := Xiaomi
   DEVICE_MODEL := Mi Router WR30U
===================================================================
+++ a/target/linux/mediatek/dts/mt7981b-xiaomi-ax3000t.dts
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi-ax3000t.dts
@@ -25,6 +25,12 @@
 	};
 };

+	chosen {
+		bootargs-append = " root=/dev/fit0 rootwait";
+		rootdisk = <&ubi_rootdisk>;
+	};
+};
+
 &pio {
 	i2c_pins: i2c-pins {
 		mux {
