--- a/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
1

--- a/target/linux/mediatek/dts/mt7981b-xiaomi-ax3000t.dts 
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi-ax3000t.dts 
2


--- a/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
3
--- a/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
@@ -12,6 +12,48 @@
 	ubimkvol /dev/ubi0 -N jffs2 -s 0x3e000
 }

+xiaomi_initial_setup()
+{
+	# initialize UBI and setup uboot-env if it's running on initramfs
+	[ "$(rootfs_type)" = "tmpfs" ] || return 0
+
+	local mtdnum="$(find_mtd_index ubi)"
+	if [ ! "$mtdnum" ]; then
+		echo "unable to find mtd partition ubi"
+		return 1
+	fi
+
+	local kern_mtdnum="$(find_mtd_index ubi_kernel)"
+	if [ ! "$kern_mtdnum" ]; then
+		echo "unable to find mtd partition ubi_kernel"
+		return 1
+	fi
+
+	ubidetach -m "$mtdnum"
+	ubiformat /dev/mtd$mtdnum -y
+
+	ubidetach -m "$kern_mtdnum"
+	ubiformat /dev/mtd$kern_mtdnum -y
+
+	if ! fw_printenv -n flag_try_sys2_failed &>/dev/null; then
+		echo "failed to access u-boot-env. skip env setup."
+		return 0
+	fi
+
+	fw_setenv -s - <<-EOF
+		boot_wait on
+		uart_en 1
+		flag_boot_rootfs 0
+		flag_last_success 1
+		flag_boot_success 1
+		flag_try_sys1_failed 8
+		flag_try_sys2_failed 8
+	EOF
+
+	fw_setenv mtdparts "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),34816k(ubi),34816k(ubi1),32768k(overlay),12288k(data),256k(KF)"
+}
+
+
 platform_do_upgrade() {
 	local board=$(board_name)

@@ -45,6 +87,11 @@
 			;;
 		esac
 		;;
+	xiaomi,mi-router-ax3000t)
+		CI_KERN_UBIPART="ubi_kernel"
+		CI_ROOT_UBIPART="ubi"
+		nand_do_upgrade "$1"
+		;;
 	cmcc,rax3000m-emmc|\
 	cmcc,xr30-emmc|\
 	glinet,gl-mt2500|\
@@ -65,6 +112,7 @@
 	esac
 }

+
 PART_NAME=firmware

 platform_check_image() {
@@ -126,8 +174,13 @@
 	local board=$(board_name)

 	case "$board" in
+
 	asus,tuf-ax4200)
 		asus_initial_setup
 		;;
+	xiaomi,mi-router-ax3000t)
+		xiaomi_initial_setup
+		;;
+
 	esac
 }
