--- a/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -493,8 +493,7 @@
 }

 function shouldDisplayAutoChannel(hwtype, configChannel, runtimeChannel) {
-	return configChannel == 'auto' || runtimeChannel == 'auto' ||
-		((isQcaWifiHwtype(hwtype) || hwtype == 'mt_dbdc') && isUnsetChannelValue(configChannel));
+    return true;
 }

 function isPlausibleTxPowerValue(txpower, hwtype) {
@@ -1880,7 +1879,7 @@
 		const ch = +channel;
 		const channelMeta = this.getChannelMeta(band, channel);
 		const bandName = String(band || '');
-		const restrictWide = (band == '5g' && !(ch > 0 && ch <= 100));
+		const restrictWide = false;

 		const filtered = [];

===================================================================
--- a/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi_mi-router.dtsi
@@ -340,10 +340,22 @@
 				read-only;
 			};

-			partition@600000 {
-				label = "ubi";
-				reg = <0x0600000 0x7000000>;
-			};
+        	// ubi_kernel is the ubi partition in stock.
+        	partition@600000 {
+	        	label = "ubi_kernel";
+		        reg = <0x0600000 0x2200000>;
+         	};
+
+        	/* ubi is the result of squashing
+        	 * consecutive stock partitions:
+        	 * - ubi1
+	         * - overlay
+        	 * - data
+        	 */
+        	partition@2800000 {
+	        	label = "ubi";
+	        	reg = <0x2800000 0x4e00000>;
+         	};

 			partition@7600000 {
 				label = "KF";
===================================================================
--- a/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
@@
 	case "$board" in
 	asus,tuf-ax4200|\
 	asus,tuf-ax6000)
 		CI_UBIPART="UBI_DEV"
 		CI_KERNPART="linux"
 		nand_do_upgrade "$1"
 		;;
+	xiaomi,mi-router-ax3000t-ubootmod)
+		fit_do_upgrade "$1"
+		;;
 	bananapi,bpi-r3|\
 	bananapi,bpi-r4|\
 	bananapi,bpi-r4-2g5|\
 	bananapi,bpi-r4-poe|\
 	bananapi,bpi-r4-lite)
 		[ -e /dev/fit0 ] && fitblk /dev/fit0
@@
 platform_check_image() {
 	local board=$(board_name)
 	local magic="$(get_magic_long "$1")"

 	[ "$#" -gt 1 ] && return 1

 	case "$board" in
+	xiaomi,mi-router-ax3000t-ubootmod)
+		fit_check_image "$1"
+		return $?
+		;;
 	bananapi,bpi-r3|\
 	bananapi,bpi-r4|\
 	bananapi,bpi-r4-2g5|\
 	bananapi,bpi-r4-poe|\
 	bananapi,bpi-r4-lite)
 		[ "$magic" != "d00dfeed" ] && {
 			echo "Invalid image type."
 			return 1
